helix-term/src/commands.rs

```diff
@@ -2934,27 +2934,40 @@ fn open(cx: &mut Context, open: Open) {
             Open::Above => cursor_line,
         };
 
+        let line_num = new_line.saturating_sub(1);
+        let text = doc.text().slice(..);
+
         // Index to insert newlines after, as well as the char width
         // to use to compensate for those inserted newlines.
         let (line_end_index, line_end_offset_width) = if new_line == 0 {
             (0, 0)
         } else {
             (
-                line_end_char_index(&doc.text().slice(..), new_line.saturating_sub(1)),
+                line_end_char_index(&text, line_num),
                 doc.line_ending.len_chars(),
             )
         };
 
+        let line_start_index = doc.text().line_to_char(line_num);
+
+        let indent_query_pos =
+            if !Range::new(line_start_index, line_end_index).is_single_grapheme(text) {
```

Comment:
I think this would be more readable if you removed the inversion and swapped the if/else branches. Also a small optimization/making the code a bit clearer (to me atleast). Simply always calculate the prev grapheme boundary and simply check if it's the sane as the line start (if it is the same then then the line only contains the newline grapheme)